syntax = "proto2";
package comm_network;

message LoadServerRequest {
  required string addr = 1;
}

message LoadServerResponse {
}

message BarrierRequest {
  required string name = 1;
  required int32 num = 2;
}

message BarrierResponse {
}

message PushKVRequest {
  required string key = 1;
  required bytes val = 2;
}

message PushKVResponse {
}

message ClearKVRequest {
  required string key = 1;
}

message ClearKVResponse {
}

message PullKVRequest {
  required string key = 1;
}

message PullKVResponse {
  required bytes val = 1;
}
